{"matches":[{"vulnerability":{"id":"GHSA-hcpj-qp55-gfph","dataSource":"https://github.com/advisories/GHSA-hcpj-qp55-gfph","namespace":"github:language:python","severity":"Critical","urls":[],"description":"GitPython vulnerable to Remote Code Execution due to improper user input validation","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-24439","epss":0.69549,"percentile":0.98559,"date":"2025-06-02"}],"fix":{"versions":["3.1.30"],"state":"fixed"},"advisories":[],"risk":59.464395},"relatedVulnerabilities":[{"id":"CVE-2022-24439","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-24439","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/gitpython-developers/GitPython/blob/bec61576ae75803bc4e60d8de7a629c194313d1c/git/repo/base.py%23L1249","https://lists.debian.org/debian-lts-announce/2023/07/msg00024.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/AV5DV7GBLMOZT7U3Q4TDOJO5R6G3V6GH/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/IKMVYKLWX62UEYKAN64RUZMOIAMZM5JN/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PF6AXUTC5BO7L2SBJMCVKJSPKWY52I5R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/SJHN3QUXPJIMM6SULIR3PR34UFWRAE7X/","https://security.gentoo.org/glsa/202311-01","https://security.snyk.io/vuln/SNYK-PYTHON-GITPYTHON-3113858","https://github.com/gitpython-developers/GitPython/blob/bec61576ae75803bc4e60d8de7a629c194313d1c/git/repo/base.py%23L1249","https://lists.debian.org/debian-lts-announce/2023/07/msg00024.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/AV5DV7GBLMOZT7U3Q4TDOJO5R6G3V6GH/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/IKMVYKLWX62UEYKAN64RUZMOIAMZM5JN/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PF6AXUTC5BO7L2SBJMCVKJSPKWY52I5R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/SJHN3QUXPJIMM6SULIR3PR34UFWRAE7X/","https://security.gentoo.org/glsa/202311-01","https://security.snyk.io/vuln/SNYK-PYTHON-GITPYTHON-3113858"],"description":"All versions of package gitpython are vulnerable to Remote Code Execution (RCE) due to improper user input validation, which makes it possible to inject a maliciously crafted remote URL into the clone command. Exploiting this vulnerability is possible because the library makes external calls to git without sufficient sanitization of input arguments.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"report@snyk.io","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-24439","epss":0.69549,"percentile":0.98559,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gitpython","version":"3.1.27"}},"found":{"versionConstraint":"<=3.1.29 (python)","vulnerabilityID":"GHSA-hcpj-qp55-gfph"},"fix":{"suggestedVersion":"3.1.30"}}],"artifact":{"id":"36b3839e614b204f","name":"gitpython","version":"3.1.27","type":"python","locations":[{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:sebastian_thiel,_michael_trier_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:gitpython:3.1.27:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gitpython@3.1.27","upstreams":[]}},{"vulnerability":{"id":"GHSA-q34m-jh98-gwm2","dataSource":"https://github.com/advisories/GHSA-q34m-jh98-gwm2","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug possible resource exhaustion when parsing file data in forms","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49767","epss":0.00974,"percentile":0.75666,"date":"2025-06-02"}],"fix":{"versions":["3.0.6"],"state":"fixed"},"advisories":[],"risk":0.60875},"relatedVulnerabilities":[{"id":"CVE-2024-49767","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-49767","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/quart/commit/5e78c4169b8eb66b91ead3e62d44721b9e1644ee","https://github.com/pallets/quart/commit/abb04a512496206de279225340ed022852fbf51f","https://github.com/pallets/werkzeug/commit/50cfeebcb0727e18cc52ffbeb125f4a66551179b","https://github.com/pallets/werkzeug/releases/tag/3.0.6","https://github.com/pallets/werkzeug/security/advisories/GHSA-q34m-jh98-gwm2","https://security.netapp.com/advisory/ntap-20250103-0007/"],"description":"Werkzeug is a Web Server Gateway Interface web application library. Applications using `werkzeug.formparser.MultiPartParser` corresponding to a version of Werkzeug prior to 3.0.6 to parse `multipart/form-data` requests (e.g. all flask applications) are vulnerable to a relatively simple but effective resource exhaustion (denial of service) attack. A specifically crafted form submission request can cause the parser to allocate and block 3 to 8 times the upload size in main memory. There is no upper limit; a single upload at 1 Gbit/s can exhaust 32 GB of RAM in less than 60 seconds. Werkzeug version 3.0.6 fixes this issue.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49767","epss":0.00974,"percentile":0.75666,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<=3.0.5 (python)","vulnerabilityID":"GHSA-q34m-jh98-gwm2"},"fix":{"suggestedVersion":"3.0.6"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"65.6.3"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"e7e499af05a9cdf5","name":"setuptools","version":"65.6.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:65.6.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@65.6.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-pr76-5cm5-w9cj","dataSource":"https://github.com/advisories/GHSA-pr76-5cm5-w9cj","namespace":"github:language:python","severity":"Critical","urls":[],"description":"GitPython vulnerable to remote code execution due to insufficient sanitization of input arguments","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-40267","epss":0.00276,"percentile":0.50878,"date":"2025-06-02"}],"fix":{"versions":["3.1.32"],"state":"fixed"},"advisories":[],"risk":0.25944},"relatedVulnerabilities":[{"id":"CVE-2023-40267","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-40267","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/gitpython-developers/GitPython/commit/ca965ecc81853bca7675261729143f54e5bf4cdd","https://github.com/gitpython-developers/GitPython/pull/1609","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/AV5DV7GBLMOZT7U3Q4TDOJO5R6G3V6GH/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PF6AXUTC5BO7L2SBJMCVKJSPKWY52I5R/","https://github.com/gitpython-developers/GitPython/commit/ca965ecc81853bca7675261729143f54e5bf4cdd","https://github.com/gitpython-developers/GitPython/pull/1609","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/AV5DV7GBLMOZT7U3Q4TDOJO5R6G3V6GH/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PF6AXUTC5BO7L2SBJMCVKJSPKWY52I5R/"],"description":"GitPython before 3.1.32 does not block insecure non-multi options in clone and clone_from. NOTE: this issue exists because of an incomplete fix for CVE-2022-24439.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-40267","epss":0.00276,"percentile":0.50878,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gitpython","version":"3.1.27"}},"found":{"versionConstraint":"<3.1.32 (python)","vulnerabilityID":"GHSA-pr76-5cm5-w9cj"},"fix":{"suggestedVersion":"3.1.32"}}],"artifact":{"id":"36b3839e614b204f","name":"gitpython","version":"3.1.27","type":"python","locations":[{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:sebastian_thiel,_michael_trier_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:gitpython:3.1.27:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gitpython@3.1.27","upstreams":[]}},{"vulnerability":{"id":"GHSA-wfm5-v35h-vwf4","dataSource":"https://github.com/advisories/GHSA-wfm5-v35h-vwf4","namespace":"github:language:python","severity":"High","urls":[],"description":"GitPython untrusted search path on Windows systems leading to arbitrary code execution","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-40590","epss":0.00325,"percentile":0.54954,"date":"2025-06-02"}],"fix":{"versions":["3.1.33"],"state":"fixed"},"advisories":[],"risk":0.24862499999999998},"relatedVulnerabilities":[{"id":"CVE-2023-40590","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-40590","namespace":"nvd:cpe","severity":"High","urls":["https://docs.python.org/3/library/subprocess.html#popen-constructor","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-wfm5-v35h-vwf4","https://docs.python.org/3/library/subprocess.html#popen-constructor","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-wfm5-v35h-vwf4"],"description":"GitPython is a python library used to interact with Git repositories. When resolving a program, Python/Windows look for the current working directory, and after that the PATH environment. GitPython defaults to use the `git` command, if a user runs GitPython from a repo has a `git.exe` or `git` executable, that program will be run instead of the one in the user's `PATH`. This is more of a problem on how Python interacts with Windows systems, Linux and any other OS aren't affected by this. But probably people using GitPython usually run it from the CWD of a repo. An attacker can trick a user to download a repository with a malicious `git` executable, if the user runs/imports GitPython from that directory, it allows the attacker to run any arbitrary commands. There is no fix currently available for windows users, however there are a few mitigations. 1: Default to an absolute path for the git program on Windows, like `C:\\\\Program Files\\\\Git\\\\cmd\\\\git.EXE` (default git path installation). 2: Require users to set the `GIT_PYTHON_GIT_EXECUTABLE` environment variable on Windows systems. 3: Make this problem prominent in the documentation and advise users to never run GitPython from an untrusted repo, or set the `GIT_PYTHON_GIT_EXECUTABLE` env var to an absolute path. 4: Resolve the executable manually by only looking into the `PATH` environment variable.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-40590","epss":0.00325,"percentile":0.54954,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gitpython","version":"3.1.27"}},"found":{"versionConstraint":"<=3.1.32 (python)","vulnerabilityID":"GHSA-wfm5-v35h-vwf4"},"fix":{"suggestedVersion":"3.1.33"}}],"artifact":{"id":"36b3839e614b204f","name":"gitpython","version":"3.1.27","type":"python","locations":[{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:sebastian_thiel,_michael_trier_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:gitpython:3.1.27:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gitpython@3.1.27","upstreams":[]}},{"vulnerability":{"id":"GHSA-xg9f-g7g7-2323","dataSource":"https://github.com/advisories/GHSA-xg9f-g7g7-2323","namespace":"github:language:python","severity":"High","urls":[],"description":"High resource usage when parsing multipart form data with many fields","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-25577","epss":0.00306,"percentile":0.53406,"date":"2025-06-02"}],"fix":{"versions":["2.2.3"],"state":"fixed"},"advisories":[],"risk":0.22949999999999998},"relatedVulnerabilities":[{"id":"CVE-2023-25577","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-25577","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/517cac5a804e8c4dc4ed038bb20dacd038e7a9f1","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-xg9f-g7g7-2323","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470","https://github.com/pallets/werkzeug/commit/517cac5a804e8c4dc4ed038bb20dacd038e7a9f1","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-xg9f-g7g7-2323","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470"],"description":"Werkzeug is a comprehensive WSGI web application library. Prior to version 2.2.3, Werkzeug's multipart form data parser will parse an unlimited number of parts, including file parts. Parts can be a small amount of bytes, but each requires CPU time to parse and may use more memory as Python data. If a request can be made to an endpoint that accesses `request.data`, `request.form`, `request.files`, or `request.get_data(parse_form_data=False)`, it can cause unexpectedly high resource usage. This allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. The amount of RAM required can trigger an out of memory kill of the process. Unlimited file parts can use up memory and file handles. If many concurrent requests are sent continuously, this can exhaust or kill all available workers. Version 2.2.3 contains a patch for this issue.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-25577","epss":0.00306,"percentile":0.53406,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<2.2.3 (python)","vulnerabilityID":"GHSA-xg9f-g7g7-2323"},"fix":{"suggestedVersion":"2.2.3"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-hrfv-mqp8-q5rw","dataSource":"https://github.com/advisories/GHSA-hrfv-mqp8-q5rw","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug DoS: High resource usage when parsing multipart/form-data containing a large part with CR/LF character at the beginning","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-46136","epss":0.0041,"percentile":0.60519,"date":"2025-06-02"}],"fix":{"versions":["2.3.8"],"state":"fixed"},"advisories":[],"risk":0.21935000000000002},"relatedVulnerabilities":[{"id":"CVE-2023-46136","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-46136","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2","https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw","https://security.netapp.com/advisory/ntap-20231124-0008/","https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2","https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw","https://security.netapp.com/advisory/ntap-20231124-0008/"],"description":"Werkzeug is a comprehensive WSGI web application library. If an upload of a file that starts with CR or LF and then is followed by megabytes of data without these characters: all of these bytes are appended chunk by chunk into internal bytearray and lookup for boundary is performed on growing buffer. This allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. This vulnerability has been patched in version 3.0.1.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8,"exploitabilityScore":2.1,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-46136","epss":0.0041,"percentile":0.60519,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<2.3.8 (python)","vulnerabilityID":"GHSA-hrfv-mqp8-q5rw"},"fix":{"suggestedVersion":"2.3.8"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-2mqj-m65w-jghx","dataSource":"https://github.com/advisories/GHSA-2mqj-m65w-jghx","namespace":"github:language:python","severity":"High","urls":[],"description":"Untrusted search path under some conditions on Windows allows arbitrary code execution","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22190","epss":0.00273,"percentile":0.50666,"date":"2025-06-02"}],"fix":{"versions":["3.1.41"],"state":"fixed"},"advisories":[],"risk":0.208845},"relatedVulnerabilities":[{"id":"CVE-2024-22190","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-22190","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/gitpython-developers/GitPython/commit/ef3192cc414f2fd9978908454f6fd95243784c7f","https://github.com/gitpython-developers/GitPython/pull/1792","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-2mqj-m65w-jghx","https://github.com/gitpython-developers/GitPython/commit/ef3192cc414f2fd9978908454f6fd95243784c7f","https://github.com/gitpython-developers/GitPython/pull/1792","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-2mqj-m65w-jghx"],"description":"GitPython is a python library used to interact with Git repositories. There is an incomplete fix for CVE-2023-40590. On Windows, GitPython uses an untrusted search path if it uses a shell to run `git`, as well as when it runs `bash.exe` to interpret hooks. If either of those features are used on Windows, a malicious `git.exe` or `bash.exe` may be run from an untrusted repository. This issue has been patched in version 3.1.41.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22190","epss":0.00273,"percentile":0.50666,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gitpython","version":"3.1.27"}},"found":{"versionConstraint":"<3.1.41 (python)","vulnerabilityID":"GHSA-2mqj-m65w-jghx"},"fix":{"suggestedVersion":"3.1.41"}}],"artifact":{"id":"36b3839e614b204f","name":"gitpython","version":"3.1.27","type":"python","locations":[{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:sebastian_thiel,_michael_trier_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:gitpython:3.1.27:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gitpython@3.1.27","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"65.6.3"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"e7e499af05a9cdf5","name":"setuptools","version":"65.6.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-65.6.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:65.6.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@65.6.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-m2qf-hxjv-5gpq","dataSource":"https://github.com/advisories/GHSA-m2qf-hxjv-5gpq","namespace":"github:language:python","severity":"High","urls":[],"description":"Flask vulnerable to possible disclosure of permanent session cookie due to missing Vary: Cookie header","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30861","epss":0.00199,"percentile":0.42671,"date":"2025-06-02"}],"fix":{"versions":["2.2.5"],"state":"fixed"},"advisories":[],"risk":0.14925},"relatedVulnerabilities":[{"id":"CVE-2023-30861","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-30861","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/flask/commit/70f906c51ce49c485f1d355703e9cc3386b1cc2b","https://github.com/pallets/flask/commit/afd63b16170b7c047f5758eb910c416511e9c965","https://github.com/pallets/flask/releases/tag/2.2.5","https://github.com/pallets/flask/releases/tag/2.3.2","https://github.com/pallets/flask/security/advisories/GHSA-m2qf-hxjv-5gpq","https://lists.debian.org/debian-lts-announce/2023/08/msg00024.html","https://security.netapp.com/advisory/ntap-20230818-0006/","https://www.debian.org/security/2023/dsa-5442","https://github.com/pallets/flask/commit/70f906c51ce49c485f1d355703e9cc3386b1cc2b","https://github.com/pallets/flask/commit/afd63b16170b7c047f5758eb910c416511e9c965","https://github.com/pallets/flask/releases/tag/2.2.5","https://github.com/pallets/flask/releases/tag/2.3.2","https://github.com/pallets/flask/security/advisories/GHSA-m2qf-hxjv-5gpq","https://lists.debian.org/debian-lts-announce/2023/08/msg00024.html","https://security.netapp.com/advisory/ntap-20230818-0006/","https://www.debian.org/security/2023/dsa-5442"],"description":"Flask is a lightweight WSGI web application framework. When all of the following conditions are met, a response containing data intended for one client may be cached and subsequently sent by the proxy to other clients. If the proxy also caches `Set-Cookie` headers, it may send one client's `session` cookie to other clients. The severity depends on the application's use of the session and the proxy's behavior regarding cookies. The risk depends on all these conditions being met.\n\n1. The application must be hosted behind a caching proxy that does not strip cookies or ignore responses with cookies.\n2. The application sets `session.permanent = True`\n3. The application does not access or modify the session at any point during a request.\n4. `SESSION_REFRESH_EACH_REQUEST` enabled (the default).\n5. The application does not set a `Cache-Control` header to indicate that a page is private or should not be cached.\n\nThis happens because vulnerable versions of Flask only set the `Vary: Cookie` header when the session is accessed or modified, not when it is refreshed (re-sent to update the expiration) without being accessed or modified. This issue has been fixed in versions 2.3.2 and 2.2.5.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30861","epss":0.00199,"percentile":0.42671,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"flask","version":"2.2.2"}},"found":{"versionConstraint":"<2.2.5 (python)","vulnerabilityID":"GHSA-m2qf-hxjv-5gpq"},"fix":{"suggestedVersion":"2.2.5"}}],"artifact":{"id":"0070c4b885fbf9a7","name":"flask","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Flask-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:flask:python-flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:flask:python_flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:flask:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:flask:flask:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/flask@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-cwvm-v4w8-q58c","dataSource":"https://github.com/advisories/GHSA-cwvm-v4w8-q58c","namespace":"github:language:python","severity":"Medium","urls":[],"description":"GitPython blind local file inclusion","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":4,"exploitabilityScore":2.6,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-41040","epss":0.00221,"percentile":0.45023,"date":"2025-06-02"}],"fix":{"versions":["3.1.37"],"state":"fixed"},"advisories":[],"risk":0.09945000000000001},"relatedVulnerabilities":[{"id":"CVE-2023-41040","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-41040","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/refs/symbolic.py#L174-L175","https://github.com/gitpython-developers/GitPython/commit/74e55ee4544867e1bd976b7df5a45869ee397b0b","https://github.com/gitpython-developers/GitPython/commit/e98f57b81f792f0f5e18d33ee658ae395f9aa3c4","https://github.com/gitpython-developers/GitPython/pull/1672","https://github.com/gitpython-developers/GitPython/releases/tag/3.1.37","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-cwvm-v4w8-q58c","https://github.com/pypa/advisory-database/tree/main/vulns/gitpython/PYSEC-2023-165.yaml","https://lists.debian.org/debian-lts-announce/2023/09/msg00036.html","https://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/refs/symbolic.py#L174-L175","https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-cwvm-v4w8-q58c","https://lists.debian.org/debian-lts-announce/2023/09/msg00036.html"],"description":"GitPython is a python library used to interact with Git repositories. In order to resolve some git references, GitPython reads files from the `.git` directory, in some places the name of the file being read is provided by the user, GitPython doesn't check if this file is located outside the `.git` directory. This allows an attacker to make GitPython read any file from the system. This vulnerability is present in https://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/refs/symbolic.py#L174-L175. That code joins the base directory with a user given string without checking if the final path is located outside the base directory. This vulnerability cannot be used to read the contents of files but could in theory be used to trigger a denial of service for the program. This issue has been addressed in version 3.1.37.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:L","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":4,"exploitabilityScore":2.6,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-41040","epss":0.00221,"percentile":0.45023,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gitpython","version":"3.1.27"}},"found":{"versionConstraint":"<3.1.37 (python)","vulnerabilityID":"GHSA-cwvm-v4w8-q58c"},"fix":{"suggestedVersion":"3.1.37"}}],"artifact":{"id":"36b3839e614b204f","name":"gitpython","version":"3.1.27","type":"python","locations":[{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/GitPython-3.1.27.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:sebastian_thiel,_michael_trier_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trierproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:sebastian_thiel,_michael_trier:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo_project:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python-gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python_gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimoproject:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:gitpython:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:byronimo:gitpython:3.1.27:*:*:*:*:*:*:*","cpe:2.3:a:python:gitpython:3.1.27:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gitpython@3.1.27","upstreams":[]}},{"vulnerability":{"id":"GHSA-q2x7-8rv6-6q7h","dataSource":"https://github.com/advisories/GHSA-q2x7-8rv6-6q7h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through indirect reference to format method","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.06592},"relatedVulnerabilities":[{"id":"CVE-2024-56326","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56326","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/48b0687e05a5466a91cd5812d604fa37ad0943b4","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-q2x7-8rv6-6q7h"],"description":"Jinja is an extensible templating engine. Prior to 3.1.5, An oversight in how the Jinja sandboxed environment detects calls to str.format allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to store a reference to a malicious string's format method, then pass that to a filter that calls it. No such filters are built-in to Jinja, but could be present through custom filters in an application. After the fix, such indirect calls are also handled by the sandbox. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.2"}},"found":{"versionConstraint":"<=3.1.4 (python)","vulnerabilityID":"GHSA-q2x7-8rv6-6q7h"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"977308c412e54ffd","name":"jinja2","version":"3.1.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-px8h-6qxv-m22q","dataSource":"https://github.com/advisories/GHSA-px8h-6qxv-m22q","namespace":"github:language:python","severity":"Low","urls":[],"description":"Incorrect parsing of nameless cookies leads to __Host- cookies bypass","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":2.6,"exploitabilityScore":1.2,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-23934","epss":0.00222,"percentile":0.45176,"date":"2025-06-02"}],"fix":{"versions":["2.2.3"],"state":"fixed"},"advisories":[],"risk":0.062160000000000014},"relatedVulnerabilities":[{"id":"CVE-2023-23934","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-23934","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pallets/werkzeug/commit/cf275f42acad1b5950c50ffe8ef58fe62cdce028","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-px8h-6qxv-m22q","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470","https://github.com/pallets/werkzeug/commit/cf275f42acad1b5950c50ffe8ef58fe62cdce028","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-px8h-6qxv-m22q","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470"],"description":"Werkzeug is a comprehensive WSGI web application library. Browsers may allow \"nameless\" cookies that look like `=value` instead of `key=value`. A vulnerable browser may allow a compromised application on an adjacent subdomain to exploit this to set a cookie like `=__Host-test=bad` for another subdomain. Werkzeug prior to 2.2.3 will parse the cookie `=__Host-test=bad` as __Host-test=bad`. If a Werkzeug application is running next to a vulnerable or malicious subdomain which sets such a cookie using a vulnerable browser, the Werkzeug application will see the bad cookie value but the valid cookie key. The issue is fixed in Werkzeug 2.2.3.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.5,"exploitabilityScore":2.1,"impactScore":1.5},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":2.6,"exploitabilityScore":1.2,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-23934","epss":0.00222,"percentile":0.45176,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<2.2.3 (python)","vulnerabilityID":"GHSA-px8h-6qxv-m22q"},"fix":{"suggestedVersion":"2.2.3"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-h5c8-rqwp-cp95","dataSource":"https://github.com/advisories/GHSA-h5c8-rqwp-cp95","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22195","epss":0.00102,"percentile":0.29274,"date":"2025-06-02"}],"fix":{"versions":["3.1.3"],"state":"fixed"},"advisories":[],"risk":0.05304000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-22195","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-22195","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/releases/tag/3.1.3","https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95","https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/","https://github.com/pallets/jinja/releases/tag/3.1.3","https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95","https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/"],"description":"Jinja is an extensible templating engine. Special placeholders in the template allow writing code similar to Python syntax. It is possible to inject arbitrary HTML attributes into the rendered HTML template, potentially leading to Cross-Site Scripting (XSS). The Jinja `xmlattr` filter can be abused to inject arbitrary HTML attribute keys and values, bypassing the auto escaping mechanism and potentially leading to XSS. It may also be possible to bypass attribute validation checks if they are blacklist-based.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":2.9,"impactScore":2.8},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22195","epss":0.00102,"percentile":0.29274,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.2"}},"found":{"versionConstraint":"<3.1.3 (python)","vulnerabilityID":"GHSA-h5c8-rqwp-cp95"},"fix":{"suggestedVersion":"3.1.3"}}],"artifact":{"id":"977308c412e54ffd","name":"jinja2","version":"3.1.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-h75v-3vvj-5mfj","dataSource":"https://github.com/advisories/GHSA-h75v-3vvj-5mfj","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25888,"date":"2025-06-02"}],"fix":{"versions":["3.1.4"],"state":"fixed"},"advisories":[],"risk":0.0442},"relatedVulnerabilities":[{"id":"CVE-2024-34064","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-34064","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/","https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/"],"description":"Jinja is an extensible templating engine. The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for CVE-2024-22195 only addressed spaces but not other characters. Accepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe. This vulnerability is fixed in 3.1.4.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25888,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.2"}},"found":{"versionConstraint":"<3.1.4 (python)","vulnerabilityID":"GHSA-h75v-3vvj-5mfj"},"fix":{"suggestedVersion":"3.1.4"}}],"artifact":{"id":"977308c412e54ffd","name":"jinja2","version":"3.1.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-gmj6-6f8f-6699","dataSource":"https://github.com/advisories/GHSA-gmj6-6f8f-6699","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through malicious filenames","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.044160000000000005},"relatedVulnerabilities":[{"id":"CVE-2024-56201","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56201","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/767b23617628419ae3709ccfb02f9602ae9fe51f","https://github.com/pallets/jinja/issues/1792","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-gmj6-6f8f-6699"],"description":"Jinja is an extensible templating engine. In versions on the 3.x branch prior to 3.1.5, a bug in the Jinja compiler allows an attacker that controls both the content and filename of a template to execute arbitrary Python code, regardless of if Jinja's sandbox is used. To exploit the vulnerability, an attacker needs to control both the filename and the contents of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates where the template author can also choose the template filename. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.2"}},"found":{"versionConstraint":">=3.0.0,<=3.1.4 (python)","vulnerabilityID":"GHSA-gmj6-6f8f-6699"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"977308c412e54ffd","name":"jinja2","version":"3.1.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-2g68-c3qc-8985","dataSource":"https://github.com/advisories/GHSA-2g68-c3qc-8985","namespace":"github:language:python","severity":"High","urls":[],"description":"Werkzeug debugger vulnerable to remote execution when interacting with attacker controlled domain","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":1.7,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34069","epss":0.0005,"percentile":0.15341,"date":"2025-06-02"}],"fix":{"versions":["3.0.3"],"state":"fixed"},"advisories":[],"risk":0.0375},"relatedVulnerabilities":[{"id":"CVE-2024-34069","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-34069","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692","https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/H4SH32AM3CTPMAAEOIDAN7VU565LO4IR/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HFERFN7PINV4MOGMGA3DPIXJPDCYOEJZ/","https://security.netapp.com/advisory/ntap-20240614-0004/","https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692","https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985","https://lists.debian.org/debian-lts-announce/2025/02/msg00026.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/H4SH32AM3CTPMAAEOIDAN7VU565LO4IR/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HFERFN7PINV4MOGMGA3DPIXJPDCYOEJZ/","https://security.netapp.com/advisory/ntap-20240614-0004/"],"description":"Werkzeug is a comprehensive WSGI web application library. The debugger in affected versions of Werkzeug can allow an attacker to execute code on a developer's machine under some circumstances. This requires the attacker to get the developer to interact with a domain and subdomain they control, and enter the debugger PIN, but if they are successful it allows access to the debugger even if it is only running on localhost. This also requires the attacker to guess a URL in the developer's application that will trigger the debugger. This vulnerability is fixed in 3.0.3.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":1.7,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34069","epss":0.0005,"percentile":0.15341,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<3.0.3 (python)","vulnerabilityID":"GHSA-2g68-c3qc-8985"},"fix":{"suggestedVersion":"3.0.3"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-w3h3-4rj7-4ph4","dataSource":"https://github.com/advisories/GHSA-w3h3-4rj7-4ph4","namespace":"github:language:python","severity":"High","urls":[],"description":"Request smuggling leading to endpoint restriction bypass in Gunicorn","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:H/A:N","metrics":{"baseScore":8.2,"exploitabilityScore":3.9,"impactScore":4.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-1135","epss":0.00044,"percentile":0.12913,"date":"2025-06-02"}],"fix":{"versions":["22.0.0"],"state":"fixed"},"advisories":[],"risk":0.03454},"relatedVulnerabilities":[{"id":"CVE-2024-1135","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-1135","namespace":"nvd:cpe","severity":"High","urls":["https://huntr.com/bounties/22158e34-cfd5-41ad-97e0-a780773d96c1","https://lists.debian.org/debian-lts-announce/2024/06/msg00027.html","https://huntr.com/bounties/22158e34-cfd5-41ad-97e0-a780773d96c1","https://lists.debian.org/debian-lts-announce/2024/06/msg00027.html","https://lists.debian.org/debian-lts-announce/2024/12/msg00018.html"],"description":"Gunicorn fails to properly validate Transfer-Encoding headers, leading to HTTP Request Smuggling (HRS) vulnerabilities. By crafting requests with conflicting Transfer-Encoding headers, attackers can bypass security restrictions and access restricted endpoints. This issue is due to Gunicorn's handling of Transfer-Encoding headers, where it incorrectly processes requests with multiple, conflicting Transfer-Encoding headers, treating them as chunked regardless of the final encoding specified. This vulnerability allows for a range of attacks including cache poisoning, session manipulation, and data exposure.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-1135","epss":0.00044,"percentile":0.12913,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gunicorn","version":"20.1.0"}},"found":{"versionConstraint":"<22.0.0 (python)","vulnerabilityID":"GHSA-w3h3-4rj7-4ph4"},"fix":{"suggestedVersion":"22.0.0"}}],"artifact":{"id":"c1d836dfd12ecfce","name":"gunicorn","version":"20.1.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:benoit_chesneau_project:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau_project:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau_project:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:gunicorn:20.1.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gunicorn@20.1.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-mq26-g339-26xf","dataSource":"https://github.com/advisories/GHSA-mq26-g339-26xf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Command Injection in pip when used with Mercurial","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}],"fix":{"versions":["23.3"],"state":"fixed"},"advisories":[],"risk":0.021},"relatedVulnerabilities":[{"id":"CVE-2023-5752","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-5752","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/","https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"],"description":"When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"22.3.1"}},"found":{"versionConstraint":"<23.3 (python)","vulnerabilityID":"GHSA-mq26-g339-26xf"},"fix":{"suggestedVersion":"23.3"}}],"artifact":{"id":"1c81af4b43f85167","name":"pip","version":"22.3.1","type":"python","locations":[{"path":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/pip-22.3.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:22.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:22.3.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@22.3.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-f9vj-2wh5-fj8j","dataSource":"https://github.com/advisories/GHSA-f9vj-2wh5-fj8j","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug safe_join not safe on Windows","cvss":[],"epss":[{"cve":"CVE-2024-49766","epss":0.00036,"percentile":0.09088,"date":"2025-06-02"}],"fix":{"versions":["3.0.6"],"state":"fixed"},"advisories":[],"risk":0.018000000000000002},"relatedVulnerabilities":[{"id":"CVE-2024-49766","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-49766","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/werkzeug/commit/2767bcb10a7dd1c297d812cc5e6d11a474c1f092","https://github.com/pallets/werkzeug/releases/tag/3.0.6","https://github.com/pallets/werkzeug/security/advisories/GHSA-f9vj-2wh5-fj8j","https://security.netapp.com/advisory/ntap-20250131-0005/"],"description":"Werkzeug is a Web Server Gateway Interface web application library. On Python < 3.11 on Windows, os.path.isabs() does not catch UNC paths like //server/share. Werkzeug's safe_join() relies on this check, and so can produce a path that is not safe, potentially allowing unintended access to data. Applications using Python >= 3.11, or not using Windows, are not vulnerable. Werkzeug version 3.0.6 contains a patch.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:H/AT:N/PR:N/UI:N/VC:L/VI:N/VA:N/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49766","epss":0.00036,"percentile":0.09088,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.2.2"}},"found":{"versionConstraint":"<=3.0.5 (python)","vulnerabilityID":"GHSA-f9vj-2wh5-fj8j"},"fix":{"suggestedVersion":"3.0.6"}}],"artifact":{"id":"72d3ead0f2583610","name":"werkzeug","version":"2.2.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Werkzeug-2.2.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.2.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.2.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.2.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-cpwx-vrp4-4pq7","dataSource":"https://github.com/advisories/GHSA-cpwx-vrp4-4pq7","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja2 vulnerable to sandbox breakout through attr filter selecting format method","cvss":[],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}],"fix":{"versions":["3.1.6"],"state":"fixed"},"advisories":[],"risk":0.017},"relatedVulnerabilities":[{"id":"CVE-2025-27516","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-27516","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/90457bbf33b8662926ae65cdde4c4c32e756e403","https://github.com/pallets/jinja/security/advisories/GHSA-cpwx-vrp4-4pq7","https://lists.debian.org/debian-lts-announce/2025/04/msg00045.html"],"description":"Jinja is an extensible templating engine. Prior to 3.1.6, an oversight in how the Jinja sandboxed environment interacts with the |attr filter allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to use the |attr filter to get a reference to a string's plain format method, bypassing the sandbox. After the fix, the |attr filter no longer bypasses the environment's attribute lookup. This vulnerability is fixed in 3.1.6.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.2"}},"found":{"versionConstraint":"<=3.1.5 (python)","vulnerabilityID":"GHSA-cpwx-vrp4-4pq7"},"fix":{"suggestedVersion":"3.1.6"}}],"artifact":{"id":"977308c412e54ffd","name":"jinja2","version":"3.1.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.2:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-hc5x-x2vx-497g","dataSource":"https://github.com/advisories/GHSA-hc5x-x2vx-497g","namespace":"github:language:python","severity":"High","urls":[],"description":"Gunicorn HTTP Request/Response Smuggling vulnerability","cvss":[{"type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6827","epss":0.00018,"percentile":0.03202,"date":"2025-06-02"}],"fix":{"versions":["23.0.0"],"state":"fixed"},"advisories":[],"risk":0.0135},"relatedVulnerabilities":[{"id":"CVE-2024-6827","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6827","namespace":"nvd:cpe","severity":"High","urls":["https://huntr.com/bounties/1b4f8f38-39da-44b6-9f98-f618639d0dd7"],"description":"Gunicorn version 21.2.0 does not properly validate the value of the 'Transfer-Encoding' header as specified in the RFC standards, which leads to the default fallback method of 'Content-Length,' making it vulnerable to TE.CL request smuggling. This vulnerability can lead to cache poisoning, data exposure, session manipulation, SSRF, XSS, DoS, data integrity compromise, security bypass, information leakage, and business logic abuse.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6827","epss":0.00018,"percentile":0.03202,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"gunicorn","version":"20.1.0"}},"found":{"versionConstraint":"<23.0.0 (python)","vulnerabilityID":"GHSA-hc5x-x2vx-497g"},"fix":{"suggestedVersion":"23.0.0"}}],"artifact":{"id":"c1d836dfd12ecfce","name":"gunicorn","version":"20.1.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/gunicorn-20.1.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:benoit_chesneau_project:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau_project:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau_project:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneauproject:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoit_chesneau:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc_project:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python-gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python_gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitcproject:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:gunicorn:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:benoitc:gunicorn:20.1.0:*:*:*:*:*:*:*","cpe:2.3:a:python:gunicorn:20.1.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/gunicorn@20.1.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"promreg-ng-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/promreg-ng-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/promreg-ng-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-03T01:32:03Z_1748923962.tar.zst?checksum=sha256%3A02a6127217518a93e0ef8739cb9f2d659109fff0d9cc25f0c13d5ca8b307a118","built":"2025-06-03T04:12:42Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-03T01:32:10Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-03T01:32:15Z","input":"xxh64:bad092583f78d4dd"},"debian":{"captured":"2025-06-03T01:32:22Z","input":"xxh64:b4717cad509f50d8"},"epss":{"captured":"2025-06-03T01:32:43Z","input":"xxh64:2f9660be834e0ce6"},"github":{"captured":"2025-06-03T01:32:21Z","input":"xxh64:cdd94f1952fcf607"},"kev":{"captured":"2025-06-03T01:32:03Z","input":"xxh64:575844c3434cc70b"},"mariner":{"captured":"2025-06-03T01:32:11Z","input":"xxh64:7628f1adb0b9b12b"},"nvd":{"captured":"2025-06-03T01:32:30Z","input":"xxh64:463c4e9a1d401101"},"oracle":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:257df3059d28d823"},"rhel":{"captured":"2025-06-03T01:33:04Z","input":"xxh64:59e7d70bf674afdc"},"sles":{"captured":"2025-06-03T01:32:46Z","input":"xxh64:3cf7f372ea0c981a"},"ubuntu":{"captured":"2025-06-03T01:33:20Z","input":"xxh64:075be8afc080f2be"},"wolfi":{"captured":"2025-06-03T01:32:31Z","input":"xxh64:1aa0b26b004bdd38"}}},"timestamp":"2025-06-03T13:10:56.043388+01:00"}}
